zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/RestoreAcceptance.java

```diff
@@ -57,6 +58,9 @@ private void takeBackup(final long backupId) {
         client.newPublishMessageCommand().messageName("name").correlationKey("key").send().join();
       }
 
+      final PartitionsActuator partitions = PartitionsActuator.of(zeebe);
+      partitions.takeSnapshot();
+
```

Comment:
:x: You should wait until the snapshot is taken before requesting backup. Snapshot is taken asynchronously, so there is a small chance that it is not included in the backup if we don't wait for it. You can use the PartitionActuator to query the snapshot.

```
  Awaitility.await("Snapshot is taken")
        .atMost(Duration.ofSeconds(60))
        .until(
            () ->
                Optional.ofNullable(partitions.query().get(1).snapshotId())
                    .flatMap(FileBasedSnapshotId::ofFileName),
            Optional::isPresent)
```